include/Yash.h

```diff
@@ -9,31 +9,38 @@
 #include <functional>
 #include <map>
 #include <numeric>
+#include <span>
 #include <string>
 #include <string_view>
 #include <vector>
 
 namespace Yash {
 
-typedef void (*CommandFunction)(const std::vector<std::string>&);
+using CommandArgs = std::span<std::string_view>;
+typedef void (*CommandFunction)(CommandArgs);
 
 struct Command {
-    std::string_view name;
-    std::string_view description;
-    CommandFunction function;
-    size_t requiredArguments;
+    const std::string_view name;
+    const std::string_view description;
+    const CommandFunction function;
+    const size_t requiredArguments;
 };
 
-template <size_t TCommandArraySize>
+struct Config {
+    const size_t maxRequiredArgs; // FIXME: Deduce based on commands - could be made constexpr
```

Comment:
I guess FIXME should be removed again??

Reply:
Yes, thanks for clarifying! Will add a comment to the config as well :+1: